default.nix

```diff
@@ -31,32 +31,30 @@ let
   addRealTimeTestLogs = drv: overrideCabal drv (attrs: {
     testTarget = "--show-details=streaming";
   });
-  plutusPkgs = (import ./pkgs { inherit pkgs; }).override {
-    overrides = self: super: {
-      plutus-prototype = addRealTimeTestLogs super.plutus-prototype;
-      language-plutus-core = doHaddockHydra (addRealTimeTestLogs super.language-plutus-core);
-    };
-  };
   cleanSourceFilter = with pkgs.stdenv;
     name: type: let baseName = baseNameOf (toString name); in ! (
       # Filter out .git repo
       (type == "directory" && baseName == ".git") ||
-      # Filter out editor backup / swap files.
       lib.hasSuffix "~" baseName ||
       builtins.match "^\\.sw[a-z]$" baseName != null ||
       builtins.match "^\\..*\\.sw[a-z]$" baseName != null ||
-
-      # Filter out locally generated/downloaded things.
-      baseName == "dist" ||
-
-      # Filter out the files which I'm editing often.
+      baseName == "dist" || baseName == "dist-newstyle" ||
+      baseName == "cabal.project.local" ||
       lib.hasSuffix ".nix" baseName ||
       lib.hasSuffix ".dhall" baseName ||
-      # Filter out nix-build result symlinks
       (type == "symlink" && lib.hasPrefix "result" baseName) ||
       (type == "directory" && baseName == ".stack-work")
     );
-  source = builtins.filterSource cleanSourceFilter ./.;
+  requiredOverlay = self: super: {
+    plutus-prototype = addRealTimeTestLogs super.plutus-prototype;
+    language-plutus-core = doHaddockHydra (addRealTimeTestLogs super.language-plutus-core);
+    mkDerivation = args: super.mkDerivation (args // optionalAttrs (args ? src) {
```

Comment:
This feels a little aggressive to me, but I guess it makes sense given the limited package set we have here. I'd probably have just put it in another function, like `doHaddockHydra` and friends.

Reply:
Also, please comment for posterity on why these things are necessary, e.g. "Ensure that all Haskell packages we build will have their source filtered"

Reply:
Actually, thinking about it, won't this also apply to all the packages in the generated package set? Are we sure it's safe for all of them?